tests/core/data_layer/test_data_rpc.py

```diff
@@ -966,16 +966,18 @@ async def process_for_data_layer_keys(
     full_node_api: FullNodeSimulator,
     data_layer: DataLayer,
     store_id: bytes32,
+    expected_value: Optional[bytes] = None,
 ) -> None:
     for sleep_time in backoff_times():
         try:
-            await data_layer.get_key_value_hash(store_id=store_id, key=expected_key)
+            value = await data_layer.get_value(store_id=store_id, key=expected_key)
         except Exception as e:
             # TODO: more specific exceptions...
             if "Key not found" not in str(e):
                 raise
         else:
-            break
+            if expected_value is None or value == expected_value:
```

Comment:
what's the expected behavior of expected_value = None but value != None ?

Reply:
If the `expected_value` is not specified (is `None`), then we don't care what the `value` is.  We only want to make sure the key is present.  If `expected_value` is specified (is `bytes`), then we want to verify that the key is present and also that the value matches `expected_value`.

Reply:
ok, on first look that seems like it will hide issues, why not just specify what is expected ?
dose this case have any use ?

Reply:
It is intentional.  It may not be the right choice and isn't a big cost to require the extra specificity at the only other use site.  There it is used when building up the store which uses only inserts so just checking key presence is sufficient.